src/modules/analysis/AnalysisViewComponent.js

```diff
@@ -0,0 +1,360 @@
+/* @flow */
+import React, { PropTypes } from 'react';
+import { I18n } from 'react-i18nify';
+import {
+  Radio,
+  PageHeader,
+  Form,
+  FormGroup,
+  Col,
+  FormControl,
+  Button,
+  Well,
+  Table,
+  Panel
+} from 'react-bootstrap';
+import FontAwesome from 'react-fontawesome';
+
+type Props = { store: any };
+
+function LabelFormat(label, md = 1) {
+  return (<Col md={md} style={{ textAlign: 'right', padding: '7px' }}><b>{label}</b></Col>);
+}
+function FieldGroup({id, label, md = 1, ...props}) {
+  return (
+    <div id={id}>
+      {LabelFormat(label, md)}
+      <Col md={2}>
+        <FormControl {... props} readOnly/>
+      </Col>
+    </div>
+  );
+}
+
+function AddButton({id, label, md, mdOffset = 0, ...props}) {
+  return (
+    <div id={id}>
+      <Col md={md} mdOffset={mdOffset}>
+        <Button {... props} disabled>
+          <FontAwesome name='plus-circle'/>{' '}
+          {label}
+        </Button>
+      </Col>
+    </div>
+  );
+}
+function NewLine() {
+  return <Form horizontal><FormGroup />
+    <hr/>
+  </Form>;
+}
+const getAnalysisTypeTerm = (store) => {
+  let val = '';
```

Comment:
I prefer to not have a mutable variable like this but return out when we have found the value. In this case we will have two return point, from `foundType.name` and where we have the return statement today with an empty value.
I'm an lazy developer so this comment goes for the other functions in this file that follow the same pattern :)